apps/ensindexer/ponder/src/ensdb-writer-worker.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * This file manages syncing ENSNode metadata:
+ * - ENSIndexer Public Config
+ * - Indexing Status
+ * into the ENSDb.
+ */
+import config from "@/config";
+
+import { secondsToMilliseconds } from "date-fns";
+import pRetry from "p-retry";
+
+import {
+  CrossChainIndexingStatusSnapshot,
+  type Duration,
+  ENSIndexerPublicConfig,
+  IndexingStatusResponseCodes,
+  OmnichainIndexingStatusIds,
+} from "@ensnode/ensnode-sdk";
+
+import { validateENSIndexerPublicConfigCompatibility } from "@/config/compatibility";
+import { EnsDbConnection, EnsDbMutation, EnsDbQuery } from "@/lib/ensdb";
+import { ensIndexerClient, waitForEnsIndexerToBecomeHealthy } from "@/lib/ensindexer";
+import { makeLogger } from "@/lib/logger";
+
+const INDEXING_STATUS_RECORD_UPDATE_INTERVAL: Duration = 1;
+
+const logger = makeLogger("ensdb-writer-worker");
+
+/**
+ * ENSDb Writer Worker
+ *
+ * Runs the following tasks:
+ * 1) On application startup, attempt to upsert serialized representation of
+ *    {@link ENSIndexerPublicConfig} into ENSDb.
+ * 2) On application startup, and then on recurring basis,
+ *    following the {@link INDEXING_STATUS_RECORD_UPDATE_INTERVAL}, attempt to
+ *    upsert serialized representation of {@link CrossChainIndexingStatusSnapshot}
+ *    into ENSDb.
+ */
+async function ensDbWriterWorker() {
+  // 0. Wait for ENSIndexer to become healthy before running the worker's logic
+  await waitForEnsIndexerToBecomeHealthy;
+
+  // 1. Create ENSDb Client
+  const ensDbConnection = new EnsDbConnection();
+  const ensDbClient = ensDbConnection.connect({
+    schemaName: config.databaseSchemaName,
+    poolConfig: {
+      connectionString: config.databaseUrl,
+    },
+  });
+
+  logger.info("ENSDb Client connected");
+
+  // 2. Create ENSDb Query object for read operations
+  const ensDbQuery = new EnsDbQuery(ensDbClient);
+  // 3. Create ENSDb Mutation object for write operations
+  const ensDbMutation = new EnsDbMutation(ensDbClient);
+
+  /**
+   * Handle ENSIndexerPublicConfig Record
+   */
+  const handleEnsIndexerPublicConfigRecord = async () => {
+    // Read stored config and in-memory config.
+    // Note: we wrap read operations in pRetry to ensure all of them are
+    // completed successfully.
+    const [storedConfig, inMemoryConfig] = await pRetry(() =>
```

Comment:
technically it think this should be 

```
await Promise.all([
  pRetry(...),
  pRetry(...)
])
```

this way if one of the reads fails, the other isn't unnecessarily re-fetched

Reply:
Awesome suggestion, thanks 🚀